c-wrapper/src/lib.rs

```diff
@@ -365,6 +467,158 @@ pub unsafe extern "C" fn free_challenge_response_session(session: *mut Challenge
     let _ = Box::from_raw(raw_session.session_wrapper as *mut ShimChallengeResponseSession);
 }
 
+/// Obtains a description of the verification API for a Veraison service running at the given base endpoint.
+///
+/// The [`VeraisonVerificationApi`] structure obtains the information needed to discover and consume the
+/// verification API effectively.
+///
+/// In order to avoid memory or resource leaks, the caller is responsible for freeing the API description
+/// structure by passing it to [`veraison_free_verification_api()`].
+///
+/// # Safety
+///
+/// It is the caller's reponsibility to ensure that `veraison_service_base_url`:
+///
+/// - is not a null pointer
+/// - points to valid, initialized data
+/// - points to memory ending in a null byte
+/// - won't be mutated for the duration of this function call
+///
+/// It is the caller's responsibility to ensure that `out_api` is
+/// not a null pointer.
+#[no_mangle]
+pub unsafe extern "C" fn veraison_get_verification_api(
+    veraison_service_base_url: *const libc::c_char,
+    out_api: *mut *mut VeraisonVerificationApi,
+) -> VeraisonResult {
+    // We have to trust the caller's char* ptr.
+    let url_str: &str = {
+        let url_cstr = CStr::from_ptr(veraison_service_base_url);
+        url_cstr.to_str().unwrap()
+    };
+
+    let api = safe_get_verification_api(url_str);
+
+    match api {
+        Ok(_) => {}
+        Err(e) => return stub_verification_api_from_error(&e, out_api),
+    }
```

Comment:
A bit shorter:
```rust
if let Err(e) = api { 
    return stub_verification_api_from_error(&e, out_api);
}
```

Reply:
That's nice - I always forget about `if let`.